test/gtest/ucp/test_ucp_worker.cc

```diff
@@ -803,6 +804,32 @@ UCS_TEST_P(test_ucp_worker_address_query, query)
 
 UCP_INSTANTIATE_TEST_CASE(test_ucp_worker_address_query)
 
+class test_ucp_worker_address_version : public ucp_test {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        add_variant_with_value(variants, UCP_FEATURE_AM, 0, "");
+    }
+};
+
+UCS_TEST_P(test_ucp_worker_address_version, pack_unpack)
+{
+    uint8_t v17_packed     = ucp_address_packed_release_version(17u);
+    uint8_t v18_packed     = ucp_address_packed_release_version(18u);
+    uint8_t current_packed = ucp_address_pack_release_version();
+
+    unsigned v17_unpacked     = ucp_address_unpack_release_version(v17_packed);
+    unsigned v18_unpacked     = ucp_address_unpack_release_version(v18_packed);
+    unsigned current_unpacked = ucp_address_unpack_release_version(current_packed);
+
+    ASSERT_EQ(17u, v17_unpacked);
+    ASSERT_EQ(18u, v18_unpacked);
```

Comment:
this check will fail after 8 releases. 
maybe add some test which would verify versions from [UCP_API_MINOR - 8, UCP_API_MINOR + 7]?

Reply:
it depends on what you want to test and may be added/replaces in separate PR, IMO full test coverage of versioning is not the target for this PR